modules/jdbc/src/main/java/org/testcontainers/jdbc/ConnectionUrl.java

```diff
@@ -0,0 +1,209 @@
+package org.testcontainers.jdbc;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.StringJoiner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/**
+ * This is an Immutable class holding JDBC Connection Url and its parsed components, used by {@link ContainerDatabaseDriver}.
+ * <p>
+ * {@link ConnectionUrl#parseUrl()} method must be called after instantiating this class.
+ *
+ * @author manikmagar
+ */
+public class ConnectionUrl {
+
+    @Getter
+    private String url;
+
+    private String databaseType;
+
+    @Getter
+    private String imageTag = "latest";
+
+    private String dbHostString;
+
+    @Getter
+    private boolean inDaemonMode = false;
+
+    @Getter
+    private Optional<String> databaseHost = Optional.empty();
+
+    @Getter
+    private Optional<Integer> databasePort = Optional.empty();
+
+    @Getter
+    private Optional<String> databaseName = Optional.empty();
+
+    @Getter
+    private Optional<String> initScriptPath = Optional.empty();
+
+    @Getter
+    private Optional<InitFunctionDef> initFunction = Optional.empty();
+
+    @Getter
+    private Optional<String> queryString;
+
+    private ConnectionUrl() {
```

Comment:
I think this is redundant, given that there is a public constructor with a parameter.

Reply:
This has been refactored along with adding a static factory method to instantiate the class.